internal/guest/storage/scsi/scsi.go

```diff
@@ -123,15 +146,35 @@ func Mount(
 		return err
 	}
 
+	// The `source` found by GetDevicePath can take some time
+	// before its actually available under `/dev/sd*`. Retry while we
+	// wait for `source` to show up.
+	for {
+		if _, err := osStat(source); err != nil {
+			if errors.Is(err, fs.ErrNotExist) || errors.Is(err, unix.ENOENT) || errors.Is(err, unix.ENXIO) {
```

Comment:
The man page for stat does mention `ENOENT` (see [here](https://man7.org/linux/man-pages/man2/lstat.2.html)), but doesn't directly mention `ENXIO`. The page could just be out of date though. 

I also did some testing where I ran Test_70LayerImagesWithNoVPmemForLayers, recommended by Amit, roughly 150 times. I only ever saw `ErrNotExist` and `ENOENT` (at the same time), but I'm not confident we'd never hit `ENXIO` so I left it.

Reply:
`syscall.ENOENT` is [implemented](https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/syscall/syscall_unix.go;l=123-124) such that it will always be equal to `fs.ErrNotExist`, so I don't think we need to check both of those. I'd also be in favor of either removing `ENXIO` from the set unless we have a reason to keep it in. Maybe @anmaxvl has context here?

Reply:
I removed the `ENOENT` option. Waiting to hear from Maksim or Amit on the `ENXIO` error.

Reply:
I think Amit was the one who was hitting `ENXIO` with newer kernel. In my case, I was running into `ENODEV`, when trying to format/encrypt scratch...